docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
 # KubeStellar Architecture
 
-**NOTE**: This is only a very rudimentary start.
+KubeStellar provides multi-cluster deployment of Kubernetes objects, controlled by a simple 
+placement policy, where Kubernetes objects are expressed in their native format with no wrapping
+or bundling. The high-level architecture for KubeStellar is illustrated in Figure 1.
 
-## Higher Level
+<figure>
+  <img src="assets/high-level-architecture.png"  alt="High Level Architecture">
+  <figcaption align="center">Figure 1 - High Level Architecture </figcaption>
+</figure>
 
-At the higher level there are three modules.
+KubeStellar relies on the concept of *spaces*, which are abstractions 
+that represent a Kubernetes API Server and its controllers. Users can use 
+spaces to perform these tasks:
 
-1. KubeStellar
-2. [the status addon](https://github.ibm.com/dettori/status-addon)
-3. KubeFlex
+1. Create *Workload Definition Spaces* (WDSs) to store the definitions of their workloads.
+2. Create *Inventory and Transport Spaces* (ITSs) to manage the inventory of clusters and 
+the transport of workloads.
+3. Register and label Workload Execution Clusters (WECs) with the Inventory and 
+Transport Space, to keep track of the available clusters and their characteristics.
+4. Define Placement Policies to specify *what" workloads and *where* should be 
+deployed on the WECs.
+5. Submit Workload Definitions in the native Kubernetes format to the WDSs, 
+and let the Placement Policies govern which WECs should receive them.
+6. Check the status of submitted workloads from the WDS.
 
-The last two are independent and KubeStellar depends on them.
+KubeStellar users can have different roles and responsibilities. 
+For example, a System Admin can set up a KubeStellar instance, 
+create WDSs and ITSs, and register WECs. An Application Owner can 
+define Placement Policies and submit Workload Definitions for deployment. 
+Other roles, such as CISO or DevOps Engineers, may also exist. However, 
+in this document, we will not make an explicit distinction between
+different user roles and will refer to a user that has all these roles.
+The user interaction with KubeStellar is described in the 
+[user interaction](./user-interaction.md) section.
 
-The status addon defines the `WorkStatus` Kind of objects and some (what?) associated stuff.
+The KubeStellar architecture has these main modules:
 
-## KubeStellar
+- *KubeStellar Controller Manager*: this module is responsible of delivering workload
+objects from the WDS to the ITS according to placement policies, and 
+updating the status of objects in the WDS.
```

Comment:
You have also been calling this thing the "kubestellar operator", right? That is the way I went in #1666 . This thing not only runs controllers but it also delivers CRD(s) into the WDS, right?

Reply:
The current name for the deployment is "kubestellar-controller-manager". The idea was to align the name of the module to the name of the deployment.

Reply:
I am OK with converging on that name. But the description here should make it clear that this thing also installs CRD(s).

Reply:
Ok, added that to the description of the kubestellar-controller-manager.

Reply:
There is a description right here, which also should mention defining the API.